app/Views/Templates/components/forms/radio.blade.php

```diff
@@ -0,0 +1,35 @@
+@props ([
+    'labelText' => '',
+    'size' => '',
+    'state' => '',
+    'checked' => '',    //'checked' or keep empty
+    'color' => '',      // e.g. bg-red-500
+    'disabledState' => ''   //'disabled' or keep empty
+])
+
+@php
+    $sizeClass = $size ? 'radio-'.$size : '';
+    $stateClass = $state ? 'radio-'.$state : '';
+    $colorClass = $color ? 'checked:'.$color : '';
+@endphp
+
+@if ($labelText)
+    <div class="form-control">
+        <label class="cursor-pointer label">
+            <span class="label-text">{{$labelText}}</span>
+            <input 
+                type="radio" 
+                {{ $checked === "checked" ? "checked='checked'" : "" }} 
```

Comment:
No need to check check states since they will be covered as part of your attribute merge. You can remove this part.